qt/scientific_interfaces/Indirect/IndirectDataAnalysisTab.h

```diff
@@ -6,152 +6,137 @@
+  void setup() override;
+  bool validate() override;
+  virtual EstimationDataSelector getEstimationDataSelector() const = 0;
+  void connectPlotPresenter();
+  void connectFitPropertyBrowser();
+  void connectDataPresenter();
+  void plotSelectedSpectra(std::vector<SpectrumToPlot> const &spectra);
+  void plotSpectrum(std::string const &workspaceName, std::size_t const &index);
+  std::string getOutputBasename() const;
+  Mantid::API::WorkspaceGroup_sptr getResultWorkspace() const;
+  std::vector<std::string> getFitParameterNames() const;
+  QList<MantidWidgets::FunctionModelDataset> getDatasets() const;
+  void enableFitButtons(bool enable);
+  void enableOutputOptions(bool enable);
+  void setPDFWorkspace(std::string const &workspaceName);
+  void updateParameterEstimationData();
+  virtual void addDataToModel(IAddWorkspaceDialog const *dialog) = 0;
+  virtual std::string getFitTypeString() const = 0;
+
+  std::unique_ptr<IndirectFitOutputOptionsPresenter> m_outOptionsPresenter;
+  Mantid::API::IAlgorithm_sptr m_fittingAlgorithm;
+  QPushButton *m_runButton;
 
 protected slots:
-  /// Slot that can be called when a user edits an input.
-  void inputChanged();
-
-  /// Plots the current preview data
-  void plotCurrentPreview();
-
-  /// Sets the selected spectrum
-  virtual void setSelectedSpectrum(int spectrum);
-
-  /// Sets the maximum spectrum
-  void setMaximumSpectrum(int spectrum);
-
-  /// Sets the minimum spectrum
-  void setMinimumSpectrum(int spectrum);
+  void setModelFitFunction();
+  void setModelStartX(double startX);
+  void setModelEndX(double endX);
+  void tableStartXChanged(double startX, WorkspaceID workspaceID, WorkspaceIndex spectrum);
+  void tableEndXChanged(double endX, WorkspaceID workspaceID, WorkspaceIndex spectrum);
+  void handleStartXChanged(double startX);
+  void handleEndXChanged(double endX);
+  void updateFitOutput(bool error);
+  void updateSingleFitOutput(bool error);
+  void fitAlgorithmComplete(bool error);
+  void singleFit();
+  void singleFit(WorkspaceID workspaceID, WorkspaceIndex spectrum);
+  void executeFit();
+  void updateParameterValues();
+  void updateParameterValues(const std::unordered_map<std::string, ParameterValue> &parameters);
+  void updateFitBrowserParameterValues(const std::unordered_map<std::string, ParameterValue> &parameters =
+                                           std::unordered_map<std::string, ParameterValue>());
+  void updateFitBrowserParameterValuesFromAlg();
+  void updateFitStatus();
+  void updateDataReferences();
+  void updateResultOptions();
+  void respondToFunctionChanged();
+
+private slots:
+  void plotSelectedSpectra();
+  void respondToSingleResolutionLoaded();
+  void respondToDataChanged();
+  void respondToDataAdded(IAddWorkspaceDialog const *dialog);
+  void respondToDataRemoved();
+  void respondToPlotSpectrumChanged();
+  void respondToFwhmChanged(double);
+  void respondToBackgroundChanged(double value);
```

Comment:
I believe the implementation for these signals are generated by the Qt framework when their class gets built

Reply:
hmm, how can QT generate the signal without knowing the function signatures?

Reply:
Sorry, I'm not sure I understand? The function signatures are the lines 141 to 143 defined in the header file

Reply:
No problem, sorry I used the wrong terms. My point is that these signals are not 'declared' in the header file. I understand that the definitions will be generated by QT. These declarations are suggested changes by me and are not in the current commit

Reply:
Ah sorry, I completely misunderstood you! I removed these signals because they are no longer used. Doing a grep for `customBoolChanged` shows it is not used on main, and I found that the other two signals could be removed when I simplified the code a bit further

Reply:
Oh, sorry, I thought they were used